src/priority/mod.rs

```diff
@@ -0,0 +1,81 @@
+use bp7::canonical::{new_canonical_block, CanonicalBlockType, CanonicalData};
+use bp7::{CanonicalBlock, EndpointID};
+use derive_try_from_primitive::TryFromPrimitive;
+use serde::de::{SeqAccess, Visitor};
+use serde::ser::{SerializeSeq, Serializer};
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PriorityError {
+    #[error("serde cbor error: {0}")]
+    Cbor(#[from] serde_cbor::Error),
+    #[error("failed to create endpoint: {0}")]
+    EndpointIdInvalid(#[from] bp7::eid::EndpointIdError),
+    #[error("invalid endpoint supplied")]
+    InvalidEndpoint,
+    #[error("payload missing")]
+    PayloadMissing,
+    #[error("invalid priority block")]
+    InvalidPriorityBlock,
+}
+
+// HOP_COUNT_BLOCK is a BlockType for a Hop Count block as defined in
```

Comment:
could you please add a comment describing the block a bit better?
don't follow my bad example :)